From 7a8b9c1d2e3f4a5b6c7d8e9f0a1b2c3d4e5f6g7h Mon Sep 17 00:00:00 2001
From: Florin Sarbu <florin@balena.io>
Date: Fri, 31 Jul 2026 15:30:00 +0000
Subject: [PATCH] 99-resin-uboot: Revert to pre-secureboot behaviour

We are updating from old 4.4.154 kernel which lacks the statx() syscall. Because of that,
lsblk from the new OS which uses statx() will fail.

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Florin Sarbu <florin@balena.io>
---
 99-resin-uboot | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/99-resin-uboot b/99-resin-uboot
index e5322d6..7c93cb5 100644
--- a/99-resin-uboot
+++ b/99-resin-uboot
@@ -22,11 +22,7 @@ else
 fi
 
 new_part=$(findmnt --noheadings --canonicalize --output SOURCE $SYSROOT -t ext4)
-if is_part_encrypted "${new_part}"; then
-    blockdev=$(lsblk -sJ "${new_part}" | jq -r '.blockdevices[].children[].name')
-else
-    blockdev=$(lsblk -sJ "${new_part}" | jq -r '.blockdevices[].name')
-fi
+blockdev=$(basename "$new_part")
 new_part_idx=$(cat "/sys/class/block/$blockdev/partition")
 
 # Convert to hex as u-boot expects hex partition numbers. Especially needed for jetson devices
-- 
2.34.1
